lucene/core/src/java/org/apache/lucene/index/IndexWriterConfig.java

```diff
@@ -459,6 +463,31 @@ public IndexWriterConfig setCommitOnClose(boolean commitOnClose) {
     return this;
   }
 
+  /**
+   * Expert: sets the amount of time to wait for merges returned by MergePolicy.findFullFlushMerges(...).
+   * If this time is reached, we proceed with the commit based on segments merged up to that point.
+   * The merges are not cancelled, and may still run to completion independent of the commit.
```

Comment:
Maybe for the last sentence: `The merges are not cancelled, and will still run to completion independent of the commit like normal segment merges`?

Maybe also state that this setting has no effect unless the `MergePolicy` actually returns merges from `findFullFlushMerges`, which the default merge policy does not?